var/da/da_radiance/da_read_obs_netcdf4ahi_jaxa.inc

```diff
@@ -510,17 +516,74 @@ end if
                end if
             end if
             num_ahi_used = num_ahi_used + 1
-            data_all = missing_r		
-
-            do k=1,nchan
-               tb = tbb(ilongitude,ilatitude,k)
-               if( tb < tbmin .or. tb > tbmax ) tb = missing_r
-               data_all(k)= tb
-            end do
-		
-! 4.0 assign information to sequential radiance structure
-!--------------------------------------------------------------------------
-            allocate ( p % tb_inv (1:nchan ))				
+
+            ibox = ilongitude/superob_width + 1
+            if ( ahi_superob_halfwidth .gt. 0 ) then
+               box_left  = superob_width * (ibox-1) +1 ! will exceed nlatitude/nlongitude if ahi_superob_halfwidth = 0
+               box_right = superob_width * ibox
+            else
+               box_left  = superob_width * (ibox-1)
+               box_right = superob_width * (ibox-1)
+            end if
+
+            ! Super-ob BT for this channel
+
+            allocate ( p % tb_inv (1:nchan) )
+            p % tb_inv = missing_r
+
+            do k = 1, nchan
+               nkeep = count(tbb(box_left:box_right,box_bottom:box_upper,k) > 0.0 )
+               temp1 = sum ( tbb(box_left:box_right,box_bottom:box_upper,k), &
+                             tbb(box_left:box_right,box_bottom:box_upper,k) > 0.0 )
+
+               if (ahi_superob_halfwidth .gt.0 .and. nkeep .gt. 0) then
+                  tb = temp1 / real(nkeep,8)
+               else
+                  ! Extract single pixel BT and radiance value for this channel
+                  tb = tbb(ilongitude,ilatitude,k)
+               end if
+
+               if (tb > 0.0)   p % tb_inv(k) = tb
+
+             end do
+
+            ! Preprocessing for cloud detection including
+            ! extracting Tb values from cloud QC buffer
+
+            if (use_clddet_zz) then
+
+               if (.not. allocated(p % superob)) then
+                  allocate( p % superob(superob_width,superob_width) )
+               end if
+
+               ! Loops over superob pixels
+               do jsup = 1, superob_width
+               do isup = 1, superob_width
+                  iysup = ilatitude + jsup-1-ahi_superob_halfwidth
+                  ixsup = ilongitude + isup-1-ahi_superob_halfwidth
+                  if (ixsup < 1 .or. ixsup > nlongitude .or. iysup < 1 .or. iysup > nlatitude ) then
+                     cycle
+                  end if
+                  if ( .not. allocated(p % superob(isup,jsup) % tb_obs ) ) then
+                     allocate ( p % superob(isup,jsup) % tb_obs  (1:nchan,1) )
+                     allocate ( p % superob(isup,jsup) % cld_qc(1) )
+                  end if
+                  p % superob(isup,jsup) % tb_obs(1:nchan,1)      = tbb( ixsup, iysup, 1:nchan )
```

Comment:
Here seems to simply assign raw tbb data into superob array. Previously superobed data is not used.

Reply:
`p % superob(isup,jsup) % tb_obs(1:nchan,1)` is for cloud detection use only, where we should retain the raw tbb value.